### PR TITLE
Support updating classification by mime types

### PR DESCRIPTION
**Pull request by PenghaiZhang**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

To support the feature of filtering classifications by MIME types, this PR updates the classification endpoint and REST module. A Jest test is also added, which requires updating testing institution `facet`.

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/api/search/EquellaSearchResource.java [hunk 1]
```diff
@@ -147,6 +148,7 @@
           @QueryParam("advancedsearch")
           String advancedSearch,
       @ApiParam(value = "An ID (not a username) of a user", required = false) @QueryParam("owner")
-          String owner);
+          String owner,
+      @ApiParam("A list of MIME types") @QueryParam("mimeTypes") List<String> mimeTypes);
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/api/search/EquellaSearchResource.java:152]: I wonder if we still need to maintain this old search endpoint.

**Comment by PenghaiZhang** [comment]
Front-end changes are quite simple so I wonder if we can include them in this PR. But I am also happy to open another PR.

**Review by edalex-ian — CHANGES_REQUESTED** [review]
So there's no explicit change to the front-end, so how does this work? Is it because we were already simply passing the whole SearchOptions into the classifications search?

Other than that query, just a couple of things to address.

**Review by edalex-ian — APPROVED** [review]
GTG! :+1:

**Pull request merged**